File: Api2.0/src/main/java/com/justanothervitor/api_2/models/payloads/response/NotePagesResponse.java
package com.justanothervitor.api_2.models.payloads.response;

import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Map;

public record NotePagesResponse<T>(
        List<T> content,int actualPage, int totalOfPages, long totalElements,
        int sizeOfPage, boolean first, boolean last,
        boolean isNull, Map<String, Object> appliedFilters){

    public static <T> NotePagesResponse<T> from(Page<T> page, Map<String, Object> appliedFilters)
    {
        return new NotePagesResponse<>(
                page.getContent(),
                page.getNumber(),
                page.getTotalPages(),
                page.getTotalElements(),
                page.getSize(),
                page.isFirst(),
                page.isLast(),
                page.isEmpty(),
                appliedFilters
        );
    }

}

File: web-front/src/app/Models/notePagesResponse.ts
/**
 * Resposta paginada da API de pesquisa de notas.
 * Espelha o NotePagesResponse.java do backend.
 */
export interface NotePagesResponse<T> {
  content: T[];
  actualPage: number;
  totalOfPages: number;
  totalElements: number;
  sizeOfPage: number;
  first: boolean;
  last: boolean;
  isNull: boolean;
  appliedFilters: Record<string, any>;
}
